### PR TITLE
Use subprocess instead of os.system()

### LINKED ISSUE #239: Replace os.system call in Python code with subprocess
I see that you are using a lot of `os.system()` in Python code. Python recommended `subprocess` functions over `os.system`.

### PR DESCRIPTION
**Pull request by prateekiiest**
Fixes #239 

`subprocess.Popen()` is strict super-set of `os.system()`

`os.system` is equivalent to Unix system command, while subprocess was a helper module created to provide many of the facilities provided by the Popen commands with an easier and controllable interface.

### CODE REVIEW

**Comment by woshikie** [comment]
Is it possible to fix the codacy issues?

**Comment by prateekiiest** [comment]
So I researched a little bit online regarding the security caveats in the subprocess documentation. I actually found some answer at stackoverflow that invoking the shell might be good for expanding glob files and other features. I was not much aware of the inside work of how the actual process takes place.
Now since you have described the whole process, I got clear about it.
And this actually got me more convinced https://stackoverflow.com/a/3172488/7516182

Thanks @hongquan for enlightening me on this. I will make the changes. It will possibly fix the codacy issues.

**Comment by hongquan** [comment]
Actually, you don't need to go to StackOverflow to know why "shell=True" is discouraged. Python official doc [explained](https://docs.python.org/3/library/subprocess.html#frequently-used-arguments) it already.

**Comment by prateekiiest** [comment]
I am sorry for the rush of commits, but since  I actually got my previous branch commits coming into this branch since I accidentally created a new branch from my previous one. Hence I got the conflicts error at first.

Just to make sure, do we need to pass separate arguments to the subprocess call or we can still pass it on as a single string and subprocess.call can parse it accordingly ?

**Comment by hongquan** [comment]
> do we need to pass separate arguments to the subprocess call or we can still pass it on as a single string and subprocess.call can parse it accordingly ?

What is this?
```py
subprocess.call('play extras/recognition-error.wav')
```
@prateekiiest DO AS PYTHON DOC!!! Why do you pass the whole line to `subprocess.call`??? Anywhere in Python Doc tell you to do that??? You are so lazy that you don't bother to split the line to make a correct code! This laziness will create bug and bug.

Didn't I explain above about the long command line with shell? Didn't I add the link to Python Doc for you to read more? `subprocess` DON'T parse the command line.

File: access_point/server/server.py [hunk 1]
```diff
@@ -13,19 +13,19 @@ def install():
 
 @app.route('/config/<stt>/<tts>/<hotword>/<wake>')
 def config(stt, tts, hotword, wake):
-    os.system('sudo bash /home/pi/SUSI.AI/susi_linux/access_point/server/config.sh {} {} {} {}'.format(stt,tts,hotword,wake))  #nosec #pylint-disable type: ignore
+    subprocess.call('sudo ./config.sh {} {} {} {}'.format(stt,tts,hotword,wake))  #nosec #pylint-disable type: ignore
```
Comment (hongquan) [inline access_point/server/server.py]: Read the doc again.

**Comment by prateekiiest** [comment]
I am really sorry about this @hongquan . I hope you understand.
I was a little unsure about the command passing in case of `subprocess`. Hence I would go through the Python Docs again and check additional materials if needed.

I will update this soon. @stealthanthrax this may be have occurred due to previous commits. I am working on reverting them. If it gets fixed then good, else I will update with a new PR as a continuation for this.

I apologize again for my mistake

**Comment by prateekiiest** [comment]
Closing this in favour of #250

**Pull request closed without merging**